core/vcpu.c

```diff
@@ -2337,6 +2338,7 @@ static void handle_cpuid_virtual(struct vcpu_t *vcpu, uint32 a, uint32 c)
     uint32 cpu_features_ext =
             feat_execute_disable |
             feat_syscall         |
+            feat_rdtscp          |
```

Comment:
1. Since this effectively enables the guest to use the `RDTSCP` and `RDPID` instructions, we should also set the `ENABLE_RDTSCP` VM-execution control (Intel SDM Vol. 3C Table 24-7 _Definitions of Secondary Processor-Based VM-Execution Controls_):

https://github.com/intel/haxm/blob/415b5e463fd7fd59ac76972213ae4808ee302b2f/core/vcpu.c#L1147

2. nit: This feature is available in Nehalem and later (i.e. all generations of Intel Core i-3/5/7) Intel CPUs (see Intel SDM Vol. 3B 17.16.2). However, officially, HAXM also supports a few pre-Nehalem models, e.g. Core 2 Duo. Therefore, it would be better to check the host CPUID for availability of this feature before exposing it to the guest. But I guess that's too much burden for you. So you could just add a TODO comment, we'll address it in a later patch.

Reply:
BTW, HAXM disables `RDTSC_EXITING`, but enables `USE_TSC_OFFSETTING` VM-execution control. According to Intel SDM Vol. 3C 25.3, `RDTSCP` won't incur a VM exit in this case.